src/components/ENSCard/ENSCardContainer/ENSCardContainer.tsx

```diff
@@ -0,0 +1,64 @@
+import { Link } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { useInView } from 'react-hook-inview';
+
+import { fetchNimiDataByENSName } from '../../../hooks/useDomainsData';
+import { useGetENSDomainsByAddress } from '../../../hooks/useGetENSDomainsByAddress';
+import { PopulatedENSCard } from '../PopulatedENSCard';
+import { ENSNameCardImage, StyledDomainName, StyledENSNameCardWrapper } from '../styleds';
+import purpleCircleURL from '../../../assets/svg/purpleCircle.svg';
+import { useENSMetadata } from '../../../hooks/useENSMetadata';
+
+type ArrElement<ArrType> = ArrType extends readonly (infer ElementType)[] ? ElementType : never;
+
+export interface ENSCardContainerProps {
+  domain: ArrElement<ReturnType<typeof useGetENSDomainsByAddress>['data']>;
+}
+
+export function ENSCardContainer({ domain }: ENSCardContainerProps) {
+  const { data: metadata, loading: metadataLoading } = useENSMetadata(domain.name!);
+  const [ref, isInView] = useInView({
+    threshold: 1,
+    defaultInView: false,
+  });
+  const [domainData, setDomainData] = useState<Awaited<ReturnType<typeof fetchNimiDataByENSName>> | null>();
+
+  // When element is in view, fetch the data
+  useEffect(() => {
+    // fetch only if not already fetched
+    if (isInView && domain?.name && domainData === undefined) {
+      fetchNimiDataByENSName(domain.name)
+        .then((data) => {
+          setDomainData(data);
+        })
+        // Ignore error
+        .catch((error) => {
+          console.log('ENSCardContainer: useEffect', error);
+          setDomainData(null);
+        });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isInView]);
```

Comment:
Suggested **React** way is listing all of the dependencies; in that case you wouldn't need to disable **ESLint**, missing are `domain.name` (which doesn't ever change) and `domainData` (whose change is covered by `if` statement).